### PR TITLE
Fixes part of #1267: Added newInstance function in TopicInfoFragment

### LINKED ISSUE #1267: Implementation change of fragment new instance
Implementation change of fragment new instance
Suggestion is:
in `ExplorationActivityPresenter`
```
   if (getExplorationManagerFragment() == null) {
      activity.supportFragmentManager.beginTransaction().add(
        R.id.exploration_fragment_placeholder,
        ExplorationManagerFragment.newInstance(internalProfileId),
        ExplorationManagerFragment.TAG
      ).commitNow()
    }
```

In `ExplorationManagerFragment`
```
  companion object {
    internal const val EXPLORATION_MANAGER_FRAGMENT_INTERNAL_PROFILE_ID_ARGUMENT_KEY = "ExplorationManagerFragment.internal_profile_id"

const val TAG = "ExplorationManagerFragment.TAG"

    fun newInstance(internalProfileId: Int): ExplorationManagerFragment {
      val explorationManagerFragment = ExplorationManagerFragment()
      val args = Bundle()
      args.putInt(EXPLORATION_MANAGER_FRAGMENT_INTERNAL_PROFILE_ID_ARGUMENT_KEY, internalProfileId)
      explorationManagerFragment.arguments = args
      return explorationManagerFragment
    }
  }

```
NAMING CONVENTIONS :
INTERNAL_PROFILE_ID_EXTRA_KEY
INTERNAL_PROFILE_ID_ARGUMENT_KEY
INTERNAL_PROFILE_ID_SAVED_KEY (saved instance state)

## Checklist
- [ ] ExplorationFragment
- [ ] TopicInfoFragment
- [ ] TopicLessonsFragment
- [ ] TopicRevisionFragment
- [ ] TopicPracticeFragment
- [ ] ExplorationManagerFragment [Currently Blocked on #1061 ]

### PR DESCRIPTION
**Pull request by sajalasati**
## Explanation

Partly fixes [#1267](https://github.com/oppia/oppia-android/issues/1267).
I have added a `newInstance` function in the `TopicInfoFragment` file as part of the code optimization suggested in the issue.
I request @vinitamurthi @nikitamarysolomanpvt @rt4914 to please this review this.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt [hunk 1]
```diff
@@ -12,6 +12,18 @@ import javax.inject.Inject
 
 /** Fragment that contains info of Topic. */
 class TopicInfoFragment : InjectableFragment() {
+  companion object{
+    /** Returns a new [TopicInfoFragment]. */
+    fun newInstance(internalProfileId: Int, topicId: String): TopicInfoFragment{
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: This function is actually not getting used anywhere. Checkout `ViewPagerAdapter` and see how can you use this function there. Also check if TopicInfoFragment is getting called from anywhere and if so, use `newInstance` accordingly.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt [hunk 2]
```diff
@@ -12,6 +12,18 @@ import javax.inject.Inject
...
+      val topicInfoFragment = TopicInfoFragment()
+      val args = Bundle()
+      args.putInt(PROFILE_ID_ARGUMENT_KEY,internalProfileId)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Nit: Space after `,` 
`args.putInt(PROFILE_ID_ARGUMENT_KEY, internalProfileId)`

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Nit space issue:
`fun newInstance(internalProfileId: Int, topicId: String): TopicInfoFragment {`

**Review by rt4914 — CHANGES_REQUESTED** [review]
This is nice but the function is not getting used anywhere, left some comments accodingly.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: `{` after TopicInfoFragment should start from next line - is this what you meant?

Comment (aggarwalpulkit596) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: No there should be a space between { and TopicInfoFragment

Comment (aggarwalpulkit596) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Also you can just ktlint -F filename to fix all these issues

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Are we not including fragment name in key? @rt4914 Can we decide the naming  convention that we may follow in different situations also?

Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Key names convention is a separate issue https://github.com/oppia/oppia-android/issues/747 and therefore its fine it does not get solved here. I was just thinking that this being a first PR i just don't to complicate things for the developer.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Done. `ViewPagerAdapter` seems to be the only file where changes were needed.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 2 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Done (Fixed the space issue)

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Done.

**Review by vinitamurthi — APPROVED** [review]
LGTM, I will defer to @rt4914 and @nikitamarysolomanpvt for the structure of this

**Comment by rt4914** [comment]
@sajalasati The Title of the PR should be `Fixes part of #1267: Added newInstance function in TopicInfoFragment`

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Looks good. Thanks.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Looks good. Thanks.

File: app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/info/TopicInfoFragment.kt]: Looks good. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM, besides the naming of argument key for example PROFILE_ID_ARGUMENT_KEY. Hope this will be handled in another PR.

**Comment by vinitamurthi** [comment]
Looks like tests pass so I am merging this PR now

**Pull request merged**